ports/lpc/boards/lpc_board_ciaa_edu_4337/src/board.c

```diff
@@ -0,0 +1,202 @@
+
+void Board_UART_Init(LPC_USART_T *pUART)
+{
+   Chip_SCU_PinMuxSet(0x7, 1, (SCU_MODE_INACT | SCU_MODE_FUNC6));                  /* P7.1 : UART2_TXD */
+   Chip_SCU_PinMuxSet(0x7, 2, (SCU_MODE_INACT | SCU_MODE_INBUFF_EN | SCU_MODE_ZIF_DIS | SCU_MODE_FUNC6));/* P7.2 : UART2_RXD */
+}
+
+/* Initialize debug output via UART for board */
+void Board_Debug_Init(void)
+{
+#if defined(DEBUG_UART)
+   Board_UART_Init(DEBUG_UART);
+   
+   Chip_UART_Init(DEBUG_UART);
+   Chip_UART_SetBaudFDR(DEBUG_UART, 115200);
+   Chip_UART_ConfigData(DEBUG_UART, UART_LCR_WLEN8 | UART_LCR_SBS_1BIT | UART_LCR_PARITY_DIS);
+   
+   /* Enable UART Transmit */
+   Chip_UART_TXEnable(DEBUG_UART);
+#endif
+}
+
+/* Sends a character on the UART */
+void Board_UARTPutChar(char ch)
+{
+#if defined(DEBUG_UART)
+   /* Wait for space in FIFO */
+   while ((Chip_UART_ReadLineStatus(DEBUG_UART) & UART_LSR_THRE) == 0) {}
+   Chip_UART_SendByte(DEBUG_UART, (uint8_t) ch);
+#endif
+}
+
+/* Gets a character from the UART, returns EOF if no character is ready */
+int Board_UARTGetChar(void)
+{
+#if defined(DEBUG_UART)
+   if (Chip_UART_ReadLineStatus(DEBUG_UART) & UART_LSR_RDR) {
+       return (int) Chip_UART_ReadByte(DEBUG_UART);
+   }
+#endif
+   return EOF;
+}
+
+/* Outputs a string on the debug UART */
+void Board_UARTPutSTR(const char *str)
+{
+#if defined(DEBUG_UART)
+   while (*str != '\0') {
+       Board_UARTPutChar(*str++);
+   }
+#endif
+}
+
+static void Board_LED_Init()
+{
+   uint32_t idx;
+
+   for (idx = 0; idx < (sizeof(gpioLEDBits) / sizeof(io_port_t)); ++idx) {
+       /* Set pin direction and init to off */
+       Chip_GPIO_SetPinDIROutput(LPC_GPIO_PORT, gpioLEDBits[idx].port, gpioLEDBits[idx].pin);
+       Chip_GPIO_SetPinState(LPC_GPIO_PORT, gpioLEDBits[idx].port, gpioLEDBits[idx].pin, (bool) false);
+   }
+}
+
+void Board_LED_Set(uint8_t LEDNumber, bool On)
+{
+   if (LEDNumber < (sizeof(gpioLEDBits) / sizeof(io_port_t)))
+        Chip_GPIO_SetPinState(LPC_GPIO_PORT, gpioLEDBits[LEDNumber].port, gpioLEDBits[LEDNumber].pin, (bool) !On);
+}
+
+bool Board_LED_Test(uint8_t LEDNumber)
+{
+   if (LEDNumber < (sizeof(gpioLEDBits) / sizeof(io_port_t)))
+       return (bool) !Chip_GPIO_GetPinState(LPC_GPIO_PORT, gpioLEDBits[LEDNumber].port, gpioLEDBits[LEDNumber].pin);
+
+   return false;
+}
+
+void Board_LED_Toggle(uint8_t LEDNumber)
+{
+   Board_LED_Set(LEDNumber, !Board_LED_Test(LEDNumber));
+}
+
+/* Returns the MAC address assigned to this board */
+void Board_ENET_GetMacADDR(uint8_t *mcaddr)
+{
+   uint8_t boardmac[] = {0x00, 0x60, 0x37, 0x12, 0x34, 0x56};
+
+   memcpy(mcaddr, boardmac, 6);
+}
+
+/* Set up and initialize all required blocks and functions related to the
+   board hardware */
+void Board_Init(void)
+{
+   /* Sets up DEBUG UART */
+   DEBUGINIT();
+
+   /* Initializes GPIO */
+   Chip_GPIO_Init(LPC_GPIO_PORT);
+
+   /* Initialize LEDs */
+   Board_LED_Init();
+   Chip_ENET_RMIIEnable(LPC_ETHERNET);
+}
+
+void Board_I2C_Init(I2C_ID_T id)
```

Comment:
@martinribelotta Are these `Board_XYZ_Init` functions needed for a minimum UART REPL? If not, I'd say let's get rid of all the clutter. Once we have consensus on a minimal REPL it will be easier to add support hardware.

Reply:
+1 The Board_* is bad idea. My direction is to provide a CSV description of pins and labels on this board. Maybe this PR need more cleanup.